client/scss/components/forms/_field-comment-control.scss

```diff
@@ -40,22 +37,45 @@ $button-size: theme('spacing.5');
     .tab-content--comments-enabled .w-field__input:focus-within > &,
     &:hover,
     &:focus,
-    &.w-field__comment-button--focused {
+    &.w-field__comment-button--focused,
+    .w-field--admin_model_chooser.w-field--commentable:hover & {
       visibility: visible;
       opacity: 1;
     }
   }
+
+  // Position comment icons in choosers beside the other actions
+  .tab-content--comments-enabled .chooser & {
+    position: unset;
+    inset-inline-end: 0;
+    margin-inline-start: 0;
+    transform: translateY(3px);
+  }
+
+  .tab-content--comments-enabled .unchosen:hover &,
+  .tab-content--comments-enabled .unchosen:focus-within &,
+  .tab-content--comments-enabled .chooser .chosen & {
+    opacity: 1;
+    visibility: visible;
+  }
 }
 
-// Add extra space to the right of commentable fields for the buttons.
 .w-field--commentable {
-  .tab-content--comments-enabled & .w-field__input {
-    padding-inline-end: $button-size;
+  .w-field__input {
+    padding-inline-end: calc($button-size + $button-space);
+  }
 
-    @include media-breakpoint-up(sm) {
-      // Partial offset to reduce the change in layout with/without comments.
-      padding-inline-end: calc($button-size / 1.5);
-    }
+  // Allow comment buttons to be placed along side the field by display flex
+  .w-field-row & {
+    margin-inline-end: 0;
+  }
+
+  // Special positioning of comment icons to sit at the top of these fields
+  textarea ~ .w-field__comment-button,
+  &.w-field--checkbox_select_multiple .w-field__comment-button,
+  &.w-field--admin_tag_widget .w-field__comment-button {
+    transform: translateY(0);
+    top: 0.3125rem;
```

Comment:
I think this should all be done without the `--commentable` selector as well if possible.

Reply:
Good call, i've moved this up and out of that block now and simplified it a bit more